scripts/test_filtering.py

```diff
@@ -0,0 +1,159 @@
+EXCLUDED_INTEGER_TESTS = [
+    "/.*integer_smart_div_param/",
+    "/.*integer_smart_rem_param/",
+    "/.*integer_default_div_param/",
+    "/.*integer_default_rem_param/",
+    "/.*_block_pbs(_base)?_param_message_[34]_carry_[34]_ks_pbs$/",
+    "~mul_crt_param_message_4_carry_4_ks_pbs",
+    "/.*test_wopbs_bivariate_crt_wopbs_param_message_[34]_carry_[34]_ks_pbs$/",
+    "/.*test_integer_smart_mul_param_message_4_carry_4_ks_pbs$/",
+    "/.*test_integer_default_add_sequence_multi_thread_param_message_4_carry_4_ks_pbs$/",
+]
+
+# skip default_div, default_rem which are covered by default_div_rem
+EXCLUDED_INTEGER_FAST_TESTS = [
+    "/.*integer_default_div_param/",
+    "/.*integer_default_rem_param/",
+    "/.*_param_message_[14]_carry_[14]_ks_pbs$/",
+    "/.*default_add_sequence_multi_thread_param_message_3_carry_3_ks_pbs$/",
+]
+
+
+def filter_integer_tests(input_args):
+    multi_bit_filter = "_multi_bit" if input_args.multi_bit else ""
+    backend_filter = ""
+    if input_args.backend == "gpu":
+        backend_filter = "gpu::"
+
+    filter_expression = [f"test(/^integer::{backend_filter}.*/)"]
+
+    if input_args.multi_bit:
+        filter_expression.append("test(~_multi_bit)")
+    else:
+        filter_expression.append("not test(~_multi_bit)")
+
+    if input_args.signed_only:
+        filter_expression.append("test(~_signed)")
+    if input_args.unsigned_only:
+        filter_expression.append("not test(~_signed)")
+
+    if input_args.fast_tests:
+        # Test only fast default operations with only two set of parameters
+        param_group = "_group_2" if input_args.multi_bit else ""
+        filter_expression.append(
+            f"test(/.*_default_.*?_param{multi_bit_filter}_message_[2-3]_carry_[2-3]{param_group}_ks_pbs/)"
+        )
+
+    excluded_tests = (
+        EXCLUDED_INTEGER_FAST_TESTS if input_args.fast_tests else EXCLUDED_INTEGER_TESTS
+    )
+    for pattern in excluded_tests:
+        filter_expression.append(f"not test({pattern})")
+
+    return " and ".join(filter_expression)
+
+
+def filter_shortint_tests(input_args):
+    multi_bit_filter, group_filter = (
+        ("_multi_bit", "_group_[0-9]") if input_args.multi_bit else ("", "")
+    )
+
+    if input_args.fast_tests:
+        msg_carry_pairs = [(2, 1), (2, 2), (2, 3)]
+    else:
+        msg_carry_pairs = [
+            (1, 1),
+            (1, 2),
+            (1, 3),
+            (1, 4),
+            (1, 5),
+            (1, 6),
+            (2, 1),
+            (2, 2),
+            (2, 3),
+            (3, 1),
+            (3, 2),
+            (3, 3),
+        ]
+        if input_args.big_instance:
+            msg_carry_pairs.append((4, 4))
+
+    filter_sub_expression = [
+        f"test(/^shortint::.*_param{multi_bit_filter}_message_{msg}_carry_{carry}{group_filter}(_compact_pk)?_ks_pbs"
+        for msg, carry in msg_carry_pairs
+    ]
+    filter_sub_expression.append("test(/^shortint::.*_ci_run_filter/)")
```

Comment:
I think integer also has some ci_run_filter stuff, but maybe we missed them before, could be added to integer as well

Reply:
as discussed probably not as integer uses exclusions instead of inclusion for shortint